src/app/business/block/bucket-detail/bucket-detail.component.ts

```diff
@@ -446,32 +455,49 @@ export class BucketDetailComponent implements OnInit {
 
   downloadFile(file) {
     let fileObjectKey;
+    let fileString: any;
+    let fileContent: any;
     if(this.folderId !=""){
       fileObjectKey = this.folderId + file.Key;
     }else{
       fileObjectKey = file.Key;
     }
-    let downloadUrl = `${this.BucketService.url}/${this.bucketId}/${fileObjectKey}`;
+    let downloadUrl = this.BucketService.url + `${this.bucketId}/${fileObjectKey}`;
     window['getAkSkList'](()=>{
-      let requestMethod = "GET";
-      let url = downloadUrl;
-      window['canonicalString'](requestMethod, url,()=>{
+      
+        let requestMethod = "GET";
+        let url = '/' + this.bucketId + '/' + fileObjectKey;
+        let requestOptions: any;
         let options: any = {};
-        this.getSignature(options);
+        requestOptions = window['getSignatureKey'](requestMethod, url);
+        options['headers'] = new Headers();
+        options = this.BucketService.getSignatureOptions(requestOptions, options);
         window['load'](file.Key,file.ETag)
         var xhr = new XMLHttpRequest();
-        xhr.open('GET', url, true);    
+        xhr.withCredentials = true;
+        xhr.open('GET', downloadUrl, true);    
         xhr.responseType = "arraybuffer";
-        xhr.setRequestHeader('Authorization', this.Signature)
-        xhr.setRequestHeader('X-Auth-Date', this.kDate)
+        xhr.setRequestHeader('Content-Type', requestOptions.headers['Content-Type']);
```

Comment:
why not it be in common function?

Reply:
Setting headers is dependent on the API call. Only a few `POST` or `PUT` calls need specific headers, else the Authorisation headers are common for all and are handled in the method `options = this.BucketService.getSignatureOptions(requestOptions, options)`

Reply:
Check app/component.ts:
Here it is...
                   xhr.setRequestHeader('Content-Type', requestOptions.headers['Content-Type']);
                    xhr.setRequestHeader('X-Auth-Token', requestOptions.headers['X-Auth-Token']);
                    xhr.setRequestHeader('X-Amz-Content-Sha256', requestOptions.headers['X-Amz-Content-Sha256']);
                    xhr.setRequestHeader('X-Amz-Date', requestOptions.headers['X-Amz-Date']);
                    xhr.setRequestHeader('Authorization', requestOptions.headers['Authorization']);
IMHO, if same code is used more than once, move it to common...Maybe in future another fucntion calling the same code

Reply:
Yes. currently there are only two places where the xhr calls are being used and I am setting the headers there manually. Object upload and Download. 
I have created an issue based on above comments #340 to track this.